dhis2_etl/tests/adx_client_tests.py

```diff
@@ -0,0 +1,33 @@
+from unittest.mock import MagicMock, patch
+
+from django.test import TestCase
+
+from dhis2_etl.adx_client import ADXClient
+from dhis2_etl.adx_transform.adx_models.adx_data import ADXMapping
+
+
+class ADXClientTests(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.adx_client = ADXClient()
+
+    def test_adx_client_config(self):
+        self.assertTrue(all((self.adx_client.host,
+                             self.adx_client.password,
+                             self.adx_client.username,
+                             self.adx_client.endpoint,
+                             self.adx_client.org_unit_id_scheme,
+                             self.adx_client.data_element_id_scheme,
+                             self.adx_client.content_type)))
+
+    @patch('dhis2_etl.adx_client.ADXClient._post')
+    def test_adx_client_post_cube(self, mock_post_method):
```

Comment:
Do you think we should extend this test and use some actual data or this is covered by other unit tests and in here we're focusing only on Client setup?

Reply:
We have tests to check ADX generation and formatting. Here I just test the client setup and request handling.